src/core/migrator.ts

```diff
@@ -98,6 +100,14 @@ export class Migrator extends Umzug implements Generator {
     });
   }
 
+  private async _validateMigrationsPathExistence() {
+      try {
+        await fs.promises.mkdir(this.migrationsPath);
+    } catch (error) {
+        console.log(error);
```

Comment:
should we stop the train here if we can't create the dir? Also, can you check if it exists first, then try to create it?

Reply:
So going through a bunch of threads regarding filesystem manipulation with nodejs. this was apparently the only solution that didnt rely on deprecated methods. 

In every scenario, mkdir would only fail if the directory already exists, so we're safe to assume that this works.

Regarding checking if it firsts exists, I can do that, but again, all solutions i could come across rely on deprecated nodejs filesystem methods. namely `fs.exists`/ `fs.existsSync`

Open to suggestions

Reply:
scratch that, i can actually use `fs.promises.access` to achieve this